### PR TITLE
tuple literal types

### LINKED ISSUE #418: Change up the game with how tuples are denoted
Variadic tuples should be like `tuple[int]`
Fixed length tuples should be `(int,)`

### LINKED ISSUE #448: Allow denotation of tuple types as tuple literals
- as a first step of #418

```py
def foo(a: int, b: str) -> (bool, 1 | 2): ...
```

### PR DESCRIPTION
**Pull request by ghost**
- resolves #448

### CODE REVIEW

File: test-data/unit/check-based-tuple-literal.test [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+reveal_type(f1) # N: Revealed type is "(int, str)"
+
+def f2() -> (str,):
+    return "sus",
+reveal_type(f2) # N: Revealed type is "def() -> (str, )"
+
+def f3() -> ():
+    return 1,  # E: Incompatible return value type (got "(int,)", expected "()")  [return-value]
+reveal_type(f3) # N: Revealed type is "def () -> ()"
+[builtins fixtures/tuple.pyi]
```
Comment (KotlinIsland) [inline test-data/unit/check-based-tuple-literal.test:20]: Can you add a test for using these types in 'value' positions?

```py
from typing import TypeVar, TypeAlias

A = TypeVar("A", bound=(int, str))

B: TypeAlias = (int, str)
```
Unfortunately, I expect these to fail. We can think about fixing this in a follow up.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 ## [Unreleased]
 ### Added
+- Deprecate python 3.7 support (#457)
+- Changed how tuples are denoted (#418)
```
Comment (KotlinIsland) [inline CHANGELOG.md:6]:
```suggestion
- Add support for literally denoted tuple types (#418)
```

File: test-data/unit/check-based-tuple-literal.test
Comment (DetachHead) [inline test-data/unit/check-based-tuple-literal.test:1]: how do u do tuples with unknown length? `(int, ...)`? add a teat for it? (if that syntax isnt allowed there should prob be a test to make sure it has a proper error)

Comment (KotlinIsland) [reply test-data/unit/check-based-tuple-literal.test:1]: Hmm, I think that would variadeic tuples are `tuple[X, ...]`, Ideally they would be `tuple[X]`, but we aren't there yet.

Test is a great idea

**Comment by KotlinIsland** [comment]
I have independently verified that the failed tests are upstream moments. we can merge

**Review by DetachHead — APPROVED** [review]

**Pull request merged**